doc/source/flexibility/binlogging_replication_improvements.rst

```diff
@@ -0,0 +1,196 @@
+==========================================================
+
+Summary of the fix:
+*******************
+
+MySQL logs DROP statements for all temporary tables irrelative of the logging
+mode under which these tables were created. This produces binlog writes and
+errand GTIDs on slaves with row and mixed logging. |Percona Server| fixes this
+by tracking the binlog format at temporary table create time and using it to
+decide whether a DROP should be logged or not.
+
+Version Specific Information
+****************************
+
+  * :rn:`5.6.38-83.0`
+    Fix implemented in |Percona Server| 5.6
+  * :rn:`5.7.17-11`
+    Fix ported to |Percona Server| 5.7
+
+Details:
+********
+
+Even with read_only mode enabled, the server permits some operations, including
+ones with temporary tables. With the previous fix, temporary table operations
+are not binlogged in row or mixed mode. But |MySQL| doesn’t track what was
+the logging mode when temporary table was created, and therefore
+unconditionally logs ``DROP`` statements for all temporary tables. These
+``DROP`` statements receive ``IF EXISTS`` addition, which is intended to make
+them harmless.
+
+|Percona Server| have fixed this with the bug fixes :psbug:`964`, upstream
+:mysqlbug:`83003`, and upstream :mysqlbug:`85258`. Morevoer, after all the
+binlogging fixes discussed so far nothing involving temporary tables is logged
+to binary log in row or mixed format, and so there is no need to consider
+``CREATE/DROP TEMPORARY TABLE`` unsafe for use in stored functions, triggers,
+and multi-statement transactions in row/mixed format. Therefore an additional
+fix was introduced to mark creation and drop of temporary tables as unsafe
+inside transactions in statement-based replication only (the fixed bug is
+:psbug:`1816`, while the correspondent upstream one is :mysqlbug:`89467` and it
+is still open).
+
+Safety of statements with a ``LIMIT`` clause
+============================================
+
+Summary of the fix:
+*******************
+
+|MySQL| considers all ``UPDATE/DELETE/INSERT ... SELECT`` statements with
+``LIMIT`` clause to be unsafe, no matter wether they are really producing
+non-deterministic result or not. |Percona Server| is more accurate, it
+acknowledges such instructions as safe when they include ``ORDER BY PK`` or
+``WHERE`` condition.
+
+Version Specific Information
+****************************
+  * :rn:`5.5.32-31.0`
+    Fix implemented in |Percona Server| 5.5
+  * :rn:`5.6.13-60.5`
+    Fix ported to |Percona Server| 5.6
+
+Details
+*******
+
+|MySQL| treats ``UPDATE/DELETE/INSERT ... SELECT`` statements with ``LIMIT``
+clause as unsafe, considering that they produce unpredictable amount of rows.
+But some such statements are still able to produce absolutely predictable
+result. One such deterministic cases takes place when statement with ``LIMIT``
+clause has an ``ORDER BY PK`` or ``WHERE`` condition.
+
+The patch making updates and deletes with a limit to be supposed as safe ones
```

Comment:
Replace the whole paragraph with

The fix has been ported from the upstream bug report :mysqlbug:`42415` (:psbug:`44`)